astropy/time/core.py

```diff
@@ -464,6 +499,49 @@ def __repr__(self):
     def __str__(self):
         return str(getattr(self, self.format))
 
+    def strftime(self, format_spec):
+        """
+        Convert Time to a string according to a format specification.
+        See time.stftime documentation for format specification.
+
+        Parameters
+        ----------
+        format_spec : string
+            Format definition of return string.
+
+        Returns
+        -------
+        string
+            Formatted string according to the given format string.
+
+        """
+
+        if self.format == 'datetime':
+            datetime_objs = list(self._time.value.flat)
+            if len(datetime_objs) == 1:
+                return datetime_objs[0].strftime(format_spec)
+            return np.array([s.strftime(format_spec) for s in datetime_objs],
+                            ).reshape(self.shape)
+
+        elif self.format not in ['iso', 'isot', 'fits', 'yday']:
+            info = list(self.replicate('iso')._time.str_kwargs())
+
+        else:
+            info = list(self._time.str_kwargs())
+
+        formatted_strings = []
+        for i in info:
```

Comment:
Untested, but i think this should work in all cases.  Note that `str_kwargs()` is a generator that automatically flattens the object and works for both scalar and array values.  This avoids creation of a temporary array.  Note also that single letter variables like `i` should generally be avoided, with the possible exception of within a list comprehension.
```
for sk in self.replicate('iso')._time.str_kwargs():
    date_tuple = date(sk['year'], sk['mon'], sk['day']).timetuple()
    .. etc

if self.isscalar:
    out = formatted_strings[0]
else:
    out = np.array(formatted_strings).reshape(self.shape)

return out
```